public/components/health-check/services/check-api.service.ts

```diff
@@ -33,11 +33,24 @@ const trySetDefault = async (checkLogger: CheckLogger) => {
             return hosts[i].id;
           }
         } catch (err) {
-          checkLogger.info(`Could not connect to API id [${hosts[i].id}]: ${err.message || err}`);
-          errors.push(`Could not connect to API id [${hosts[i].id}]: ${err.message || err}`);
+          checkLogger.info(
+            `Could not connect to API id [${hosts[i].id}]: ${
+              err.message || err
+            }`,
+          );
+          errors.push(
+            `Could not connect to API id [${hosts[i].id}]: ${
+              err.message || err
+            }`,
+          );
         }
       }
       if (errors.length) {
+        if (errors[0].includes('ERROR3099 - 405: Method Not Allowed')) {
```

Comment:
**nitpick**: this is only checking the first element of the array, In the case, the user has more Wazuh API host entries set, if the API with the mismatched version related to the change of HTTP method to authenticate is not placed the first one, it could not be caught.

I think the error should be managed for each Wazuh API host. We could add more information related to the error instead of the `Could not connect to API id [host ID]` message.